extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -79,16 +87,27 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void initialize() {
         GatewayConfig configuration = getConfigAs(GatewayConfig.class);
-        if (configuration.host == null || configuration.host.isEmpty()) {
+        if (Strings.isNullOrEmpty(configuration.host)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "Host must be specified in the configuration!");
             return;
         }
-        if (configuration.code == null || configuration.code.isEmpty()) {
+        if (Strings.isNullOrEmpty(configuration.code)) {
+            if(Strings.isNullOrEmpty(configuration.username) || Strings.isNullOrEmpty(configuration.userkey)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security code must be provided in the configuration when initializing for first time!");
```

Comment:
So a valid config actually needs the security code AND a username AND a pre-shared key?

Reply:
No - valid config requires either a security code OR - both username and a pre-shared key

Reply:
1. So the message should reflect this.
2. Reading the IKEA docs, username seems to be chosen by the application. I don't think it makes sense to have it as a config parameter at all.

Reply:
What do you mean by 'chosen by the application' ?

If it's chosen by the handler (as it currently is) - it also needs to be set in the Dslt object. - So if not stored as a thing configuration - no way to know what's it after system (or binding) restart.

From my tests - one username can be authenticated ONLY ONCE - after that each call to 15011/9063 doesn't work anymore unless different username is provided